nautobot/ui/src/utils/string.js

```diff
@@ -4,3 +4,23 @@ export function toTitleCase(str, separator = " ") {
         .map((x) => (x ? x[0].toUpperCase() + x.slice(1) : ""))
         .join(" ");
 }
+
+export function slugify(str) {
+    let data = str.replace(/^\s+|\s+$/g, ""); // trim
+    data = data.toLowerCase();
+
+    // Remove accents, swap ñ for n, etc
+    var from = "àáäâèéëêìíïîòóöôùúüûñç·/_,:;";
+    var to = "aaaaeeeeiiiioooouuuunc------";
+    for (var i = 0, l = from.length; i < l; i++) {
+        data = data.replace(new RegExp(from.charAt(i), "g"), to.charAt(i));
+    }
+
+    data = data
+        .replace(/[^a-z0-9 -]/g, "") // remove invalid chars
+        .replace(/\s+/g, "-") // collapse whitespace and replace by -
+        .replace(/-+/g, "-") // collapse dashes
+        .replace(/_+/g, "-"); // replace underscores with hyphens
```

Comment:
Ok yeah, I'm definitely not a fan of this function. It's both overly complex and also overly naive - there's all sorts of non-ASCII characters that it doesn't handle, it feels really inefficient (*how* many regexp replaces are performed each time this function is invoked?), and again I think it relies on assumptions about app route names that aren't necessarily going to be true in all cases.